### PR TITLE
Feature/slop 61/connect 'I watched it' and 'I want it' buttons

### PR DESCRIPTION
**Pull request by fdlai**
## Describe your changes
**Issue:** Previously, the **I watched it** and **I want it** buttons did not actually make any change to the backend. Therefore, their state wasn't saved upon refreshing the page, and the profile section wouldn't render anything for **slops to gobble**.

**Changes:** When the movie modal mounts, I have a useQuery to fetch to the server to get what slops a user has watched and wants to watch. Then, when clicking a button, I have a useMutation to update the server accordingly.

## Issue ticket number and link
slop-61
https://tripleten-apiary.atlassian.net/jira/software/projects/SLOP/boards/2?selectedIssue=SLOP-61

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/routes/profile-route/profile-sidebar.jsx [hunk 1]
```diff
@@ -42,6 +42,7 @@ export const ProfileSidebar = () => {
     setToken(null);
     setIsLoggedIn(false);
     client.resetStore();
+    setCurrentUser(null);
```
Comment (fdlai) [inline src/routes/profile-route/profile-sidebar.jsx:45]: Fixes a bug, where when logging out, the currentUser is not reset. Which meant that, after logging out, I could still click on the movie modal, and see if the last logged in user has clicked either of those 2 buttons.

File: src/store/client-context-provider.jsx [hunk 2]
```diff
@@ -28,7 +28,24 @@ export const ClientContextProvider = ({ children }) => {
     setClient(
       new ApolloClient({
         link: authLink.concat(httpLink),
-        cache: new InMemoryCache(),
+        cache: new InMemoryCache({
```
Comment (fdlai) [inline src/store/client-context-provider.jsx:31]: Prevents extra fetching, when the cache doesn't know how to merge the 'watched' or 'wishlist' fields.

**Comment by dani9401** [comment]
Hey @fdlai - Awesome job on this! I found one thing that may need updated or we may decide it belongs in a new ticket but take a look at the attached video and let me know what you think. https://github.com/jahorwitz/slopopedia/assets/126746048/b65478c2-d039-43cd-b036-33e4a8cab508

**Review by ATribbey — APPROVED** [review]
The buttons are working as described! It was quite helpful looking over how you solved it, as it has helped me understand the underlying concepts a bit more.

Well done Fred!

**Comment by fdlai** [comment]
> Hey @fdlai - Awesome job on this! I found one thing that may need updated or we may decide it belongs in a new ticket but take a look at the attached video and let me know what you think. https://github.com/jahorwitz/slopopedia/assets/126746048/b65478c2-d039-43cd-b036-33e4a8cab508

I didn't notice that. That's a good idea! Thanks. I'll work on that.

**Comment by fdlai** [comment]
Per @dani9401 's suggestion, I've now updated this branch so that using the movie-preview modal within the profile section, and clicking the **I watched it** or **I want it** buttons, immediately reflect those changes in the profile section. This branch could use another review please!

**Review by skovakina — APPROVED** [review]
Hi! I tested the changes and I like it. When I click 'I want it' or 'I watched it,' it shows the changes immediately. Approve!

File: src/routes/profile-route/profile-sidebar.jsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ import sidebarHeart from "../../images/sidebar-heart.svg";
 import sidebarWrench from "../../images/sidebar-wrench.svg";
 
 export const ProfileSidebar = () => {
-  const { setIsLoggedIn } = useCurrentUser();
+  const { setIsLoggedIn, setCurrentUser } = useCurrentUser();
```
Comment (tomershvueli) [inline src/routes/profile-route/profile-sidebar.jsx:11]: We're not using the new variable we're bringing in, let's remove this.

**Review by tomershvueli — APPROVED** [review]
This is working really well! See 1 minor comment, and also we need to sync up with `main` and resolve conflicts before I can merge this.

File: src/routes/profile-route/profile-sidebar.jsx (hunk 3 above)
Comment (fdlai) [reply src/routes/profile-route/profile-sidebar.jsx]: Ok, I've updated the branch!

File: src/store/client-context-provider.jsx [hunk 4]
```diff
@@ -28,39 +29,7 @@ export const ClientContextProvider = ({ children }) => {
     setClient(
       new ApolloClient({
         link: authLink.concat(httpLink),
-        cache: new InMemoryCache({
-          // Custom merge function
```
Comment (fdlai) [inline src/store/client-context-provider.jsx:32]: Moved this object to **constants.js** to clean up the code

**Pull request merged**